silo/views.py

```diff
@@ -958,12 +965,35 @@ def siloDetail(request, silo_id):
     data = []
     query = makeQueryForHiddenRow(json.loads(silo.rows_to_hide))
 
+    """
+    TODO: Add highlight in view template to the specific read that failed 
+    Note:    There is a chance a service gets stuck in "tasks_running" if a service worker terminates unexpectedly and the task id
+    could not be removed from the task.
+    """
+
+    tasks_running = 0
+    tasks_failed = 0
+
+    for read in silo.reads.all():
```

Comment:
* If you just wanna check the `task_id`, it's better to use `values_list` method.
* Is there no better way to get this info instead of fetching from the db and looping through it, maybe using the `count` function?

Reply:
Agree. Database-wise this is too inefficient. Also, you don't need to know the quantity. You just need to know if there is any task running or errored.